prof.py

```diff
@@ -0,0 +1,40 @@
+import cvxpy as cp
+import numpy as np
+from scipy import sparse
+from scipy.sparse import linalg as splinalg
+import time
+
+import diffcp.cone_program as cone_prog
+import diffcp.cones as cone_lib
+import diffcp.utils as utils
+
+
+m = 100
+n = 50
+
+A, b, c, cone_dims = utils.least_squares_eq_scs_data(m, n)
+for mode in ["lsqr", "dense", "sparse"]:
+    x, y, s, derivative, adjoint_derivative = cone_prog.solve_and_derivative(
```

Comment:
The profiling times here ignore the different pre-computation times hidden in here, which I think are non-negligible between the sparse/dense versions, perhaps #8 is a good solution here?